### PR TITLE
Make CI consistent with other Scipp projects

### LINKED ISSUE #149: Update project to use same setup as scippnexus / scippneutron
Now that scippneutron provides a PyPI package we should be able to use a tox-based setup and make PyPI packages as well. Mantid is required in some places so approach may be closer to what is implemented in scippneutron.

### PR DESCRIPTION
**Pull request by nvaytet**
- Use `tox` for static analysis
- Use `setup.cfg` and `pyproject.toml`, remove `setup.py`
- Build on ubuntu, macos & windows

Fixes #149

### CODE REVIEW

File: .github/workflows/pr_and_main.yml [hunk 1]
```diff
@@ -19,44 +19,53 @@ defaults:
...
-    env:
-      DOCS_HTML_DIR: ${{ github.workspace }}/docs_html
+    name: build ${{ matrix.os }}, py${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-22.04", "macos-11", "windows-2019"]
+        python-version: ["3.8"]
+    # env:
```
Comment (jl-wynen) [inline .github/workflows/pr_and_main.yml]: Remove?

File: setup.cfg [hunk 2]
```diff
@@ -1,3 +1,43 @@
+[metadata]
```
Comment (jl-wynen) [inline setup.cfg:1]: Most of these can be moved to pyproject.toml. See, e.g., https://github.com/scipp/scippneutron/blob/main/pyproject.toml
 A few fields are not supported yet. But while you are at it, you could move everything you can at this point.

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+[build-system]
+requires = [
+  "setuptools>=42",
+  "setuptools_scm[toml]>=6.2",
+]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools_scm]
+
+[tool.pytest.ini_options]
```
Comment (jl-wynen) [inline pyproject.toml:10]: scippneutron is a bit stricter now (https://github.com/scipp/scippneutron/blob/b717619590c29922852f4c15c291aa413e8f483e/pyproject.toml#L43) and I think we should use these options in all projects.

File: setup.cfg (hunk 2 above)
Comment (nvaytet) [reply setup.cfg:1]: In Plopp and scippnexus they are in the setup. Is it better to have in the pyproject?

Comment (jl-wynen) [reply setup.cfg:1]: As I understand it, the plan (by the Python devs) is to move everything into pyproject.toml and remove setup.(cfg|py) completely. So eventually, we have to move. But we still have years left.

File: tox.ini [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+[testenv:static]
+description = Code formatting and static analysis
+basepython = python3.8
+skip_install = true
+deps = -r requirements/static.txt
+allowlist_externals = find
+commands =
+  nbstripout --extra-keys 'metadata.language_info.version cell.metadata.jp-MarkdownHeadingCollapsed' --drop-empty-cells .
+  yapf --recursive --in-place src tests
+  flake8 .
```
Comment (SimonHeybrock) [inline tox.ini:18]: I have also added a `deps` env in some projects:

https://github.com/scipp/scipp/blob/31868a165bc52785551be2a22648d1f1368d6fa3/tox.ini#L72-L78

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**